### PR TITLE
Testlab1 2

### PR DESCRIPTION
**Pull request by sidey383**

### CODE REVIEW

File: testlab1-2.c [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    char pattern[] = "0123456789abcdef";
+    if (!in) {
+        printf("can't create in.txt. No space on disk?\n");
+        return -1;
+    }
+    printf("Creating large text with seed %u... ", seed[bigTestN]);
+    fflush(stdout);
+    srand(seed[bigTestN]);
+    t = GetTickCount();
+    if(fputs("0123456789abcdef\n", in) == EOF) {
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:86]: need one space after `if`, `for`, etc. -- globally

File: testlab1-2.c [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        printf("can't write in in.txt. No space on disk?\n");
+        return -1;
+    }
+    unsigned int r;
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:90]: should be inside for-loop.
r -> randomPrefixLength

File: testlab1-2.c [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        printf("FAILED\n");
+        testN++;
+        return 1;
+    }
+}
+
+static int feederBigRand1(void)
+{
+    FILE *const in = fopen("in.txt", "w");
+    DWORD t;
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:75]: please merge with `t = GetTickCount()` -- globally

File: testlab1-2.c [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+        testN++;
+        return -1;
+    }
+    for(unsigned int i = 0; i < 16; i++) {
+        int n;
+        if(ScanInt(out, &n) != Pass) {
+            passed = 0;
+            printf("wrong output -- ");
+            break;
+        }else if(n != 0) {
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c]: need one space before `else`

File: testlab1-2.c [hunk 5]
```diff
@@ -0,0 +1,304 @@
+#include "testLab.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+static int testN = 0;
+static int seed[] = {1234u, 383u};
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:6]: `static const int seeds`

File: testlab1-2.c [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+    printf("Creating large text with seed %u... ", seed[bigTestN]);
+    fflush(stdout);
+    srand(seed[bigTestN]);
+    t = GetTickCount();
+    if(fputs("0123456789abcdef\n", in) == EOF) {
+        printf("can't write in in.txt. No space on disk?\n");
+        return -1;
+    }
+    unsigned int r;
+    for(unsigned int i = 1; i < 1024*1024*8; i++) {
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:91]: need one space before, one after each arithmetic operator
change postfix ++ to prefix ++ -- globally

File: testlab1-2.c [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+    LabTimeout = (int)t*10;
+    fflush(stdout);
+    return 0;
+}
+
+static int CheckerBigRand1(void)
+{
+    FILE *const out = fopen("out.txt", "r");
+    unsigned int passed = 1;
+    unsigned int bigTestN = testN - sizeof(testInOut) / sizeof(testInOut[0]);
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:122]: if bigTestN < 0 or bigTestN >= size of seeds, then print a message about internal error, and quit

File: testlab1-2.c [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+        testN++;
+        return 1;
+    }
+}
+
+static int FeederBigRand1(void)
+{
+    FILE *const in = fopen("in.txt", "w");
+    DWORD t;
+    unsigned int bigTestN = testN - sizeof(testInOut) / sizeof(testInOut[0]);
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:76]: if bigTestN < 0 or bigTestN >= size of seeds, then print a message about internal error, and quit

File: testlab1-2.c [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+    return 0;
+}
+
+static int CheckFromArray(void)
+{
+    FILE *const out = fopen("out.txt", "r");
+    int i, passed = 1;
+    if (!out) {
+        printf("can't open out.txt\n");
+        testN++;
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:46]: ++testN -- globally in this tester

File: testlab1-2.c [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+        return 1;
+    }
+}
+
+static int FeederBig(void)
+{
+    FILE *const in = fopen("in.txt", "w");
+    DWORD t;
+    char str[] = "abcdabcd\n";
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:179]: `const`

File: testlab1-2.c [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+        return 1;
+    }
+}
+
+static int FeederBigRand1(void)
+{
+    FILE *const in = fopen("in.txt", "w");
+    DWORD t;
+    unsigned int bigTestN = testN - sizeof(testInOut) / sizeof(testInOut[0]);
+    char pattern[] = "0123456789abcdef";
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:77]: `const`

File: testlab1-2.c [hunk 12]
```diff
@@ -0,0 +1,304 @@
...
+static int CheckFromArray(void)
+{
+    FILE *const out = fopen("out.txt", "r");
+    int i, passed = 1;
+    if (!out) {
+        printf("can't open out.txt\n");
+        testN++;
+        return -1;
+    }
+    for (i = 0; i < testInOut[testN].n; i++) {
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:49]: int i = 0

File: testlab1-2.c [hunk 13]
```diff
@@ -0,0 +1,304 @@
...
+    }
+    fprintf(in, "%s", testInOut[testN].in);
+    fclose(in);
+    return 0;
+}
+
+static int CheckFromArray(void)
+{
+    FILE *const out = fopen("out.txt", "r");
+    int i, passed = 1;
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:43]: remove `i` from here

File: testlab1-2.c [hunk 14]
```diff
@@ -0,0 +1,304 @@
...
+    }
+    if (passed) {
+        passed = !HaveGarbageAtTheEnd(out);
+    }
+    fclose(out);
+    if (passed) {
+        printf("PASSED\n");
+        testN++;
+        return 0;
+    } else {
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:168]: need space before `else`

File: testlab1-2.c [hunk 15]
```diff
@@ -0,0 +1,304 @@
...
+            printf("wrong output -- ");
+            break;
+        }
+    }
+    if(passed) {
+        srand(seed[bigTestN]);
+        unsigned int number = 1;
+        unsigned int a, b;
+        for (unsigned int i = 1; i < 1024 * 1024 * 8; i++) {
+            unsigned int r = ((unsigned int) rand()) % 17;
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:141]: r -> randomPrefixLength

**Review by Evgueni-Petrov-aka-espetrov — CHANGES_REQUESTED** [review]

File: testlab1-2.c (hunk 9 above)
Comment (sidey383) [reply testlab1-2.c:46]: other tests use both testN++ and ++testN, but most of all testN++
Also I can see something like [this](https://github.com/Evgueni-Petrov-aka-espetrov/TestDriver/blob/39e2debacd9ab2511fa6b502a759ec31edc0d762/testlab4.c#L103) and [this](https://github.com/Evgueni-Petrov-aka-espetrov/TestDriver/blob/39e2debacd9ab2511fa6b502a759ec31edc0d762/testlab4.c#L116)

File: testlab1-2.c [hunk 16]
```diff
@@ -0,0 +1,300 @@
...
+    }
+}
+
+static int FeederBigRand1(void) {
+    FILE *const in = fopen("in.txt", "w");
+    unsigned int bigTestN = testN - sizeof(testInOut) / sizeof(testInOut[0]);
+    if (bigTestN >= sizeof(seed) / sizeof(seed[0])) {
+        printf("\nInternal error: incorrect test number\n");
+        ++testN;
+        return -1;
```
Comment (Evgueni-Petrov-aka-espetrov) [inline testlab1-2.c:76]: need to close in.txt on error. here and below.

**Review by Evgueni-Petrov-aka-espetrov — CHANGES_REQUESTED** [review]

**Review by Evgueni-Petrov-aka-espetrov — APPROVED** [review]

**Pull request merged**